### PR TITLE
tests: use load generator to generate load in perf tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**
It includes:
 . set the load generator for performance tests
 . new parameters to configure performance tests
 . a new test to check snap connections for snap provided interfaces 
 . a new test to check many installations in parallel.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -1060,12 +1060,36 @@ suites:
...
+                # and also to make sure it is not blocking spread scripts
+                snap install --edge --devmode test-snapd-load-generator
+                tests.systemd create-and-start-unit load-generator "/snap/bin/test-snapd-load-generator.run -d $LOAD_DURATION -l $CPU_LOAD"
+                tests.systemd wait-for-service -n 5 --wait 1 --state active load-generator
+            fi
         restore-each: |
+            if [ -n "$CPU_LOAD" ]; then
+                # Now we make sure all the load generator processes are killed before
+                # the load-generator service is removed
+                #shellcheck disable=SC2009
```
Comment (olivercalder) [inline spread.yaml:1083]: Leftover shellchecks (below as well)? Should these be enabled, or removed, or put behind some variable?

File: tests/lib/snaps/store/test-snapd-load-generator/load-generator [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+-d#!/usr/bin/env python3
```
Comment (olivercalder) [inline tests/lib/snaps/store/test-snapd-load-generator/load-generator:1]: I've never seen the `-d` prefix to a shebang, is this a typo or does it do something intentional?

File: tests/perf/interfaces-snap-provided/task.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    "$TESTSTOOLS"/snaps-state install-local "$PROVIDER_SNAP"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+
+    echo "For each snap-provided slot from $PROVIDER_SNAP"
+    SNAP_MOUNT_DIR="$(os.paths snap-mount-dir)"
+    for slotcmd in "$SNAP_MOUNT_DIR"/bin/"$PROVIDER_SNAP".* ; do
+        slotcmd_bn=$(basename "$slotcmd")
+        slot_iface=$(echo "$slotcmd_bn" | tr '.' ':')
+
+        #shellcheck disable=SC2001
```
Comment (olivercalder) [inline tests/perf/interfaces-snap-provided/task.yaml:27]: Another leftover shellcheck?

File: tests/perf/parallel-installs/task.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+restore: |
+    snap set system experimental.parallel-instances=null
+
+execute: |
+    if [ -z "$NUM_PARALLEL" ]; then
+        NUM_PARALLEL=20
+    fi
+
+    # shellcheck disable=SC2086
```
Comment (olivercalder) [inline tests/perf/parallel-installs/task.yaml:21]: Another few here

**Review by olivercalder — APPROVED** [review]
Looks good to me, thanks! Just a few tiny notes

**Review by Meulengracht — APPROVED** [review]
lgtm, but agree with olivers comments

**Pull request merged**